ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -123,189 +116,189 @@ public SECDataConverter(string rawSource, string destination, string knownEquity
-                                );
-                                var reports = tickerReports.GetOrAdd(
-                                    report.Report.FilingDate.Date,
-                                    _ => new List<ISECReport>()
-                                );
-
-                                reports.Add(report);
-                            }
+                            parsingText = false;
                         }
-                    );
 
-                    Parallel.ForEach(Reports.Keys, ticker =>
+                        // To ensure that we can serialize/deserialize data with hours, minutes, seconds
+                        if (currentTagName == "FILING-DATE" || currentTagName == "PERIOD" ||
+                            currentTagName == "DATE-OF-FILING-CHANGE" || currentTagName == "DATE-CHANGED")
                         {
-                            List<ISECReport> reports;
-                            if (!Reports[ticker].TryRemove(currentDate, out reports))
-                            {
-                                return;
-                            }
+                            newTextLine = $"{newTextLine.TrimEnd()} 00:00:00";
+                        }
 
-                            WriteReport(reports, ticker);
+                        // Encode all contents inside tags to prevent errors in XML parsing.
+                        // The json deserializer will convert these values back to their original form
+                        if (!parsingText && HasValue(newTextLine))
+                        {
+                            newTextLine =
+                                $"<{currentTagName}>{SecurityElement.Escape(GetTagValueFromLine(newTextLine))}</{currentTagName}>";
+                        }
+                        // Escape all contents inside TEXT tags
+                        else if (parsingText)
+                        {
+                            newTextLine = SecurityElement.Escape(newTextLine);
                         }
-                    );
 
-                    // This will clean up after ourselves without having to pay
-                    // the expense of deleting every single file inside the raw_data folder
-                    Directory.Delete(extractDataPath, true);
+                        // Don't encode the opening tag
+                        if (currentTagName == "TEXT")
+                        {
+                            parsingText = true;
+                        }
+
+                        xmlText.AppendLine(newTextLine);
+                    }
+
+                    ISECReport report;
+                    try
+                    {
+                        report = factory.CreateSECReport(xmlText.ToString());
+                    }
+                    // Ignore unsupported form types for now
+                    catch (DataException)
+                    {
+                        return;
```

Comment:
Do we want to log this case at least? This same comment applies for some other exception handlers as well.

Thanks for providing the comment regarding the source of the `DateException` 👍